app/models/game.rb

```diff
@@ -39,4 +39,45 @@ def piece_at(column_coordinate, row_coordinate)
     pieces.where(x_position: column_coordinate, y_position: row_coordinate).first
   end
 
+  def check?(color)
+    king = pieces.find_by(type: 'King', color: color)
+    if color == "black"
+      opposite_color = "white"
+    else
+      opposite_color = "black"
+    end
+    #find pieces still on board with opposite color (captured pieces are destroyed)
+    opponents_pieces = pieces.where(color: opposite_color)
+    #for each of opponents pieces check whether can move to position of king
+    check = false
+    opponents_pieces.each do |piece|
+      if piece.valid_move?(king.x_position, king.y_position)
```

Comment:
this will shorten your code: http://ruby-doc.org/core-2.2.3/Enumerable.html#method-i-find